src/h5web/dataset-visualizer/vis/MatrixVis.tsx

```diff
@@ -13,34 +14,162 @@ interface Props {
   data: HDF5Value;
 }
 
+type CellProps = Omit<GridChildComponentProps, 'data'>;
+
+interface MatrixProps {
+  children: CellRendererFunc;
+  width: number;
+  height: number;
+}
+
+interface IndexCellProps {
+  index: number;
+  style: React.CSSProperties;
+}
+
+type CellRendererFunc = (props: GridChildComponentProps) => JSX.Element;
+
 function MatrixVis(props: Props): JSX.Element {
   const { dims, data } = props;
+  const gridDimensions = {
+    rowHeight: 32,
+    rowCount: dims[0] + 1,
+    columnWidth: 116,
+    columnCount: (dims.length === 2 ? dims[1] : 1) + 1,
+  };
```

Comment:
I am conflicted here. I choose to add 1 to `rowCount` and `columnCount` because the final grid has 1 more row and 1 more column. But it may also be possible to have `rowCount` and `columnCount` to be equal to the exact number of rows and columns of the *true* data and move the `+1` in the `Index` components.

Reply:
Hmm, yeah tough one. I think it's easier to keep it here, since `react-window` does need to calculate the total width and height of the grid, index cells included.